src/Symfony/Component/Uid/UuidV6.php

```diff
@@ -61,6 +64,11 @@ public static function generate(\DateTimeInterface $time = null, Uuid $node = nu
             self::$node = sprintf('%06x%06x', ($seed[0] ^ $node[1]) | 0x010000, $seed[1] ^ $node[2]);
         }
 
-        return $uuid.self::$node;
+        $composedUuid = $uuid.self::$node;
+        if ('' === $composedUuid) {
```

Comment:
this should not be necessary here. Concatenating 2 non-empty strings cannot produce an empty string.

Reply:
I agree and disagree!

At this point, `$uuid` contains the output from `substr` and `self::$node` the output of `sprintf`.

We as humans can see that the usage of both of these methods will not result in an empty string. This is where I agree that the concatenation of these two values cannot produce an empty string.

Both `substr` and `sprintf`, by contract, return a string. This could be an empty string. By contact, these functions offer no guarantees that the output will not be an empty string. Looking solely at the return types of the functions that generate the values that are concatenated, an empty string can be the result. This is, I believe, how phpstan would interpret things. I will pop together a simple phpstan example to check ...

Reply:
No. At this point, `$uuid` contains the result of concatenating the output of 5 different `substr` and the static string `'-'` and the static string `-6`. There is 2 non-empty static strings in that concatenation. So even if all `substr` calls return an empty string, `$uuid` cannot be empty.

Reply:
Yes, you are correct. I was just confirming for myself since I don't have a particularly effective static analyser in my head.

Reply:
I'll also check other changes to be sure that emptiness checks are not entirely superfluous.

Reply:
Well ok then! I'll remove more code.